Aptoide/aptoide-client-v8
# Bugfix/mob 88

**What does this PR do?**

This PR fixes a concurrency issue that was breaking the retrieving of the google advertising id. This also makes the method to retrieve the id reactive and refactors it's usages to reflect this.

**Database changed?**

No

**Where should the reviewer start?**

- [x] IdsRepository.java

**How should this be manually tested?**

This touched upon most of the app. I'd say the best way to make sure everything is okay is to check the requests and make sure the id being sent matches the one you see on logs;

**What are the relevant tickets?**

  Tickets related to this pull-request: [MOB-88](https://aptoide.atlassian.net/browse/MOB-88)

**Code Review Checklist**
